tensorboard/components/vz_line_chart2/vz-line-chart2.ts

```diff
@@ -266,6 +266,53 @@ Polymer({
     '_outliersChanged(ignoreYOutliers, _chart)'
   ],
 
+  ready: function() {
+    this.scopeSubtree(this.$.tooltip, true);
+    this.scopeSubtree(this.$.chartdiv, true);
+  },
+
+  attached() {
+    const option = {capture: true, passive: true};
+    this._listen(this, 'mousedown', this._onMouseDown.bind(this), option);
+    this._listen(this, 'mouseup', this._onMouseUp.bind(this), option);
+    this._listen(window, 'keydown', this._onKeyDown.bind(this), option);
+    this._listen(window, 'keyup', this._onKeyUp.bind(this), option);
+  },
+
+  detached() {
+    this.cancelAsync(this._makeChartAsyncCallbackId);
+    if (this._chart) this._chart.destroy();
+    if (this._listeners) {
+      this._listeners.forEach(({node, eventName, func, option}) => {
+        node.removeEventListener(eventName, func, option);
+      });
+      this._listeners.clear();
+    }
+  },
+
+  _listen(node: Node, eventName: string, func: (event) => void, option = {}) {
```

Comment:
Fair enough, though it seems like `passive` isn't really critical except for touch events (which these aren't).  How important is setting `capture` to true, i.e. is it a correctness issue or a performance one and if the latter, does it have a noticeable impact?  It would be nice not to have to reinvent the wheel unless we really have to.

Reply:
`passive` is more than touch events. It tells browser that JS should not block drawing thread. Anyways, yes, `capture` guarantees the correctness as any node can stop propagation and cause this logic to incorrectly behave.

Reply:
Ok.  Could you add a comment then above the definition of options, that notes that we can't use Polymer's listener logic because we want to set those options?